src/redis-benchmark.c

```diff
@@ -238,40 +240,56 @@ void _serverAssert(const char *estr, const char *file, int line) {
     *((char*)-1) = 'x';
 }
 
-static redisConfig *getRedisConfig(const char *ip, int port,
-                                   const char *hostsocket)
+static redisContext *getRedisContext(const char *ip, int port,
+                                     const char *hostsocket)
 {
-    redisConfig *cfg = zcalloc(sizeof(*cfg));
-    if (!cfg) return NULL;
-    redisContext *c = NULL;
-    redisReply *reply = NULL, *sub_reply = NULL;
+    redisContext *ctx = NULL;
+    redisReply *reply =  NULL;
     if (hostsocket == NULL)
-        c = redisConnect(ip, port);
+        ctx = redisConnect(ip, port);
     else
-        c = redisConnectUnix(hostsocket);
-    if (c == NULL || c->err) {
+        ctx = redisConnectUnix(hostsocket);
+    if (ctx == NULL || ctx->err) {
         fprintf(stderr,"Could not connect to Redis at ");
-        char *err = (c != NULL ? c->errstr : "");
+        char *err = (ctx != NULL ? ctx->errstr : "");
         if (hostsocket == NULL) fprintf(stderr,"%s:%d: %s\n",ip,port,err);
         else fprintf(stderr,"%s: %s\n",hostsocket,err);
-        goto fail;
```

Comment:
here you may have a potentical memory leak, if ctx is not null, you can use redisFree(ctx) before return to resolve this